test/sequential/test-util-debug.js

```diff
@@ -23,34 +23,43 @@
 const common = require('../common');
 const assert = require('assert');
 
-if (process.argv[2] === 'child')
-  child();
+const [, , modeArgv, sectionArgv] = process.argv;
+
+if (modeArgv === 'child')
+  child(sectionArgv);
 else
   parent();
 
 function parent() {
-  test('foo,tud,bar', true);
-  test('foo,tud', true);
-  test('tud,bar', true);
-  test('tud', true);
-  test('foo,bar', false);
-  test('', false);
+  test('foo,tud,bar', true, 'tud');
+  test('foo,tud', true, 'tud');
+  test('tud,bar', true, 'tud');
+  test('tud', true, 'tud');
+  test('foo,bar', false, 'tud');
+  test('', false, 'tud');
+
+  test('###', true, '###');
+  test('hi:)', true, 'hi:)');
+  test('f$oo', true, 'f$oo');
+  test('f$oo', false, 'f.oo');
+  test('no-bar-at-all', false, 'bar');
 }
 
-function test(environ, shouldWrite) {
+function test(environ, shouldWrite, section) {
   let expectErr = '';
-  if (shouldWrite) {
-    expectErr = 'TUD %PID%: this { is: \'a\' } /debugging/\n' +
-                'TUD %PID%: number=1234 string=asdf obj={"foo":"bar"}\n';
-  }
   const expectOut = 'ok\n';
 
   const spawn = require('child_process').spawn;
-  const child = spawn(process.execPath, [__filename, 'child'], {
+  const child = spawn(process.execPath, [__filename, 'child', section], {
     env: Object.assign(process.env, { NODE_DEBUG: environ })
   });
 
-  expectErr = expectErr.split('%PID%').join(child.pid);
+  if (shouldWrite) {
+    const SECTION = section.toUpperCase();
```

Comment:
You could probably even get away with just using `section.toUpperCase()` and `child.pid`. They're only used to create an error message in a test.